devdoc/rc_ptr.md

```diff
@@ -0,0 +1,34 @@
+# `rc_ptr` requirements
+
+## Overview
+
+`rc_ptr` is a module that encapsulates reference-counted pointers.
+
+## Exposed API
+```c
+typedef void (*RC_PTR_FREE_FUNC)(void* context);
+
+typedef struct RC_PTR_TAG
+{
+    const void* ptr;
+    RC_PTR_FREE_FUNC free_func;
+} RC_PTR;
+
+THANDLE_TYPE_DECLARE(RC_PTR);
+
+#define PRI_RC_PTR "p"
+
+#define RC_PTR_VALUE(rc) ((rc)->ptr)
+
+#define RC_PTR_AS_VOIDPTR(rc) (((rc) == NULL) ? NULL : (rc)->ptr)
+
+MOCKABLE_FUNCTION(, THANDLE(RC_PTR), rc_ptr_create_with_move_memory, const void*, ptr, RC_PTR_FREE_FUNC, free_func);
```

Comment:
>const [](http://example.com/codeflow?start=69&length=5)

Why does it need to be "const"?

Reply:
This module is based on `rc_string`. 

```c
MOCKABLE_FUNCTION(, THANDLE(RC_STRING), rc_string_create_with_move_memory, const char*, string);
```

has `const`, so I made it `const`.  I think it makes sense because it tells the user that this module will not modify this memory.